tests/audiohelper.js

```diff
@@ -124,6 +156,24 @@ describe('AudioHelper', () => {
       });
     });
 
+    describe('navigator.permissions', () => {
+      it('should listen for microphone state changes', () => {
+        let onChangeHandler = sinon.stub();
+        let microphonePermissionStatus = {
+          state: 'prompt',
+          addEventListener: (eventName, listener) => {
+            if (eventName === 'change') {
+              listener()
+            }
+          }
+        };
+        navigator.permissions.query = () => Promise.resolve(microphonePermissionStatus);
+        audio = new AudioHelper(onActiveOutputsChanged, onActiveInputChanged, {});
```

Comment:
It seems like this test might not going through SDK codepaths, unless my understanding is incorrect. If you remove these two lines, I don't think much changes? It seems like you're creating the `microphonePermissionStatus` object above and then lines `172~173` test that code path?

Reply:
I see your point :( . I am having a hard time testing this feature correctly. Any suggestions?
cc: @charliesantos

Reply:
Let's get the implementation approved then I'll review the tests 😄 
That way, we don't have to keep updating the tests while we review the PR.